drivers/rtc/rtc_max31329.c

```diff
@@ -0,0 +1,554 @@
+#ifdef CONFIG_RTC_ALARM
+	.alarm_get_supported_fields = max31329_alarm_get_supported_fields,
+	.alarm_set_time = max31329_alarm_set_time,
+	.alarm_get_time = max31329_alarm_get_time,
+	.alarm_is_pending = max31329_alarm_is_pending,
+	.alarm_set_callback = max31329_alarm_set_callback,
+#endif
+};
+
+int max31329_init(const struct device *dev)
+{
+	const struct max31329_config *config = dev->config;
+	int ret;
+	uint8_t reg_value;
+
+	/* Check if the I2C bus is ready */
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("Failed to get pointer to %s device!", config->i2c.bus->name);
+		return -EINVAL;
+	}
+
+	/* Check if the device is alive by reading the status register */
+	ret = i2c_reg_read_byte_dt(&config->i2c, MAX31329_REG_STATUS, &reg_value);
+	if (ret) {
+		LOG_ERR("Failed to read from MAX31329! (err %i)", ret);
+		return -EIO;
+	}
+
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_INT_EN, 0x00);
+	if (ret) {
+		LOG_ERR("Failed to disable interrupts! (err %i)", ret);
+		return ret;
+	}
+
+	/* Enable power fail threshold based on DT properties */
+	reg_value = config->pwer_fail_threshold ? config->pwer_fail_threshold << 2 : 0;
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_PWR_MGMT, reg_value);
+	if (ret) {
+		LOG_ERR("Failed to configure power fail! (err %i)", ret);
+		return ret;
+	}
+
+	/* Enable trickle charging based on DT properties */
+	reg_value = config->trickle_charging_settings;
+	reg_value |= config->trickle_charging_settings ? BIT(7) : 0;
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_TRICKLE_REG, reg_value);
+	if (ret) {
+		LOG_ERR("Failed to configure trickle charging! (err %i)", ret);
+		return ret;
+	}
+
+	/* Enable Digital-In (Event) if specified in DT */
+	if (config->digital_in_enable) {
+		ret = i2c_reg_read_byte_dt(&config->i2c, MAX31329_REG_INT_EN, &reg_value);
+		if (ret) {
+			LOG_ERR("Failed to read INT_EN register! (err %i)", ret);
+			return ret;
+		}
+		/* Enable DIE (Digital-In Event) */
+		reg_value |= BIT(3);
+		ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_INT_EN, reg_value);
+		if (ret) {
+			LOG_ERR("Error when enabling Digital-In (DIE): %i", ret);
+			return ret;
+		}
+	}
+
+#ifdef MAX31329_INT_GPIOS_IN_USE
+	struct max31329_data *data = dev->data;
+
+	data->callback_work = callback_work;
+
+#endif
+
+	LOG_INF("%s is initialized!", dev->name);
+	return 0;
+}
+
+#define MAX31329_INIT(inst)                                                                        \
+	static const struct max31329_config max31329_config_##inst = {                             \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.digital_in_enable = DT_INST_PROP(inst, digital_in_enable),                        \
+		.pwer_fail_threshold = DT_INST_ENUM_IDX_OR(inst, pvft, 0),                         \
+		.trickle_charging_settings = DT_INST_ENUM_IDX_OR(inst, trickle, 0),                \
+		IF_ENABLED(MAX31329_INT_GPIOS_IN_USE,						   \
```

Comment:
Alignment here is a bit off, this can happen to macros if clang-format is used :)